kale/evaluate/uncertainty_metrics.py

```diff
@@ -0,0 +1,827 @@
+            inner_list_errors = []
+
+            for id_ in inner_list_bin:
+                inner_list_errors.append(list([val for key, val in true_errors_lm.items() if str(key) == str(id_)])[0])
+
+            pred_bins_errors.append(inner_list_errors)
+            pred_bins_keys.append(inner_list_bin)
+
+        bins_acc = []
+        bins_sizes = []
+        # key_groups = []
+        for q in range((num_bins)):
+            inner_bin_correct = 0
+
+            inbin_errors = pred_bins_errors[q]
+
+            for error in inbin_errors:
+                if q == 0:
+                    lower = 0
+                    upper = fold_bounds[q]
+
+                    if error <= upper and error > lower:
+                        inner_bin_correct += 1
+
+                elif q < (num_bins) - 1:
+                    lower = fold_bounds[q - 1]
+                    upper = fold_bounds[q]
+
+                    if error <= upper and error > lower:
+                        inner_bin_correct += 1
+
+                else:
+                    lower = fold_bounds[q - 1]
+                    upper = 999999999999999999999999999999
+
+                    if error > lower:
+                        inner_bin_correct += 1
+
+            if inner_bin_correct == 0:
+                accuracy_bin = 0.0
+            elif len(inbin_errors) == 0:
+                accuracy_bin = 1.0
+            else:
+                accuracy_bin = inner_bin_correct / len(inbin_errors)
+            bins_sizes.append(len(inbin_errors))
+            bins_acc.append(accuracy_bin)
+
+            all_qs_perc[q].append(accuracy_bin)
+            all_qs_size[q].append(len(inbin_errors))
+            all_qs_errorbound_concat_targets_sep[i_lm][q].append(accuracy_bin)
+
+        # Weighted average over all bins
+        weighted_mean_lm = 0.0
+        total_weights = 0.0
+        for l_idx in range(len(bins_sizes)):
+            bin_acc = bins_acc[l_idx]
+            bin_size = bins_sizes[l_idx]
+            weighted_mean_lm += bin_acc * bin_size
+            total_weights += bin_size
+        weighted_av = weighted_mean_lm / total_weights
+        all_target_perc.append(weighted_av)
+
+    # Weighted average for each of the quantile bins.
+    weighted_av_binwise = []
+    for binidx in range(len(all_qs_perc)):
+        bin_accs = all_qs_perc[binidx]
+        bin_asizes = all_qs_size[binidx]
+
+        weighted_mean_bin = 0.0
+        total_weights_bin = 0.0
+        for l_idx in range(len(bin_accs)):
+            b_acc = bin_accs[l_idx]
+            b_siz = bin_asizes[l_idx]
+            weighted_mean_bin += b_acc * b_siz
+            total_weights_bin += b_siz
+
+        # Avoid div by 0
+        if weighted_mean_bin == 0 or total_weights_bin == 0:
+            weighted_av_bin = 0.0
+        else:
+            weighted_av_bin = weighted_mean_bin / total_weights_bin
+        weighted_av_binwise.append(weighted_av_bin)
+
+    # No weighted average, just normal av
```

Comment:
av? what's av? be careful. Also for below.

Reply:
are you advising i change variable names that have av to average e.g.  normal_av_bin_wise -> normal_average_bin_wise ?

Reply:
ave might be a good compromise: normal_ave_bin_wise

Reply:
changed occurances of av to ave